frontend/src/react-query/api/schema.tsx

```diff
@@ -0,0 +1,189 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { config } from 'config';
+import type {
+  SchemaRegistryCompatibilityMode,
+  SchemaRegistryConfigResponse,
+  SchemaRegistrySubject,
+  SchemaRegistrySubjectDetails,
+} from 'state/restInterfaces';
+
+export const useListSchemasQuery = () => {
+  return useQuery<SchemaRegistrySubject[]>({
+    queryKey: ['schemaRegistry', 'subjects'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/subjects`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema subjects');
+      }
+
+      const data = await response.json();
+      // could also be a "not configured" response
+      if (!Array.isArray(data)) {
+        return [];
+      }
+
+      return data;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaModeQuery = () => {
+  return useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'mode'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/mode`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema mode');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.mode ?? null;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaCompatibilityQuery = () => {
+  return useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'compatibility'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/config`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema compatibility');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.compatibility ?? null;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaDetailsQuery = (subjectName: string, options?: { enabled?: boolean }) => {
```

Comment:
You can make `subjectName` optional so then we don't need `|| ''` fallback.

Also I would suggest always using an object so that argument order does not matter.